### PR TITLE
Added error for no active session

### PR DESCRIPTION
**Pull request by franciscop**
Added error check for no active session. Generate early error and return in that situation. Following talk on #126:

> Because if a token is incoming and a new secret is also being generated, there is certainly no way that the token will match the secret, so even trying to match is likely a waste and I figure providing a different error in that case will be better than the single error.

@dougwilson

Edit: this is WIP and haven't built+tested it locally. Some extra tests would be needed which I'll try to make later on when I can clone+build+test locally.

### CODE REVIEW

**Comment by franciscop** [comment]
@dougwilson please let me know if this is okay so I can continue with the testing

File: index.js [hunk 1]
```diff
@@ -101,14 +101,21 @@ function csurf (options) {
       return token
     }
 
+    var submitted = value(req)
```
Comment (dougwilson) [inline index.js]: I would continue the refactoring around this in order to not call the `value` function if the method is ignored, similar to the current behavior.

File: index.js [hunk 2]
```diff
@@ -101,14 +101,21 @@ function csurf (options) {
...
+
     // generate & set secret
     if (!secret) {
+      if (!ignoreMethod[req.method] && submitted) {
+        return next(createError(403, 'csrf token submitted but there is no active session', {
```
Comment (dougwilson) [inline index.js]: I would either make the message much less specific, or add additional logic and errors. For example, I feel like this message has the following two issues:

1. This module has the ability to be used with the double-submit cookie pattern. In this case, the secret is not stored in a session at all, yet this error will still be triggered, likely misleading users, especially those who are using double submit cookies and also have a session system in their code.
2. The term "active session" may have different meaning to different people. I get a lot of folks at the `express-session` module, for example, who think "active session" is only when they set a user or other property on the session, but that is of course not the case -- simply having the session cookie is what makes it active.

If the logic around this is just as-is, a suggestion could be to make the error be `'no establish csrf secret'` and maybe like `ENOECSRFTOKEN` or such. This is just a suggestion.

Comment (franciscop) [inline index.js]: Makes a lot of sense, you have of course a much better understanding of csurf while I just know the *session* side of it. Will fix it.

**Comment by franciscop** [comment]
@dougwilson please let me know if now it's correct or whether I should change anything else. Thank you so much for helping/guiding me so far.

**Pull request closed without merging**